txstratum/prometheus.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+import os
+from typing import TYPE_CHECKING, Dict
+
+from prometheus_client import CollectorRegistry, Gauge, write_to_textfile  # type: ignore
+
+if TYPE_CHECKING:
+    from txstratum.manager import TxMiningManager
+
+
+# Define prometheus metrics and it's explanation
+METRIC_INFO = {
```

Comment:
It's probably good to add a comment here that these need to be a subset of `manager.status()`.

It might also be good to have a comment on `manager.status()` explaining this, otherwise someone might change some metrics names and it'll break things here.

Reply:
I changed it and added a simple test. Can you review it again, please?

Reply:
To be honest, I think it became more complex now, with the lambdas on `METRIC_INFO`. I think the old way was easier to understand. Maybe I'm not seeing something.

I did like the idea of having a test for it, instead of relying on the comments, as I suggested. If we decide to go back to the old way, we could just check that `METRIC_INFO.keys()` is a subset of `manager.status().keys()`.

Reply:
I changed it because the `manager.status()` was calculating more things than needed, e.g. generating the status of the miners. This way, we can explicitly add whatever metric we want regardless of the public status.

Reply:
Ah, makes sense!

Reply:
Another possibility would be to create a `get_metrics()` method in the manager. So, the PrometheusExporter would just call it. What do you think?

Reply:
That might be a bit cleaner than using the lambdas on `METRIC_INFO`.

Reply:
Done!